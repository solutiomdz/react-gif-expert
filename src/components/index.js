export * from './AddCategory';
export * from './GifGrids';
export * from './GifItem';
