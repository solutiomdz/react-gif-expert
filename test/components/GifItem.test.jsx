import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas sobre GifItem.jsx', () => { 

    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';
    
    test('Debe hacer match contra el SnapShot', () => {

        const {container} = render( <GifItem title={ title } url={ url } />)
        expect (container).toMatchSnapshot();
    });


    test('Debe mostrar la iagen con el URL y el ALT indicado', () => { 
        render (<GifItem title={ title } url={ url}/>);
        // screen.debug();

        // expect( screen.getByRole('img').src).toBe( url );
        // expect( screen.getByRole('img').alt).toBe( title );
        // Esto mismo se puede hacer desestructurando screen.getAllByRole
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );

    });

    test('Debe mostrar el tirulo en el componente', () => { 
        render( <GifItem  title={ title } url={ url }/>);
        expect( screen.getByText( title ) ).toBeTruthy();

    });
});