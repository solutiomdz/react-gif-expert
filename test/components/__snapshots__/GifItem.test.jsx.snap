// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifItem.jsx Debe hacer match contra el SnapShot 1`] = `
<div>
  <div
    class="="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
